### PR TITLE
Add Port Support

### PR DESCRIPTION
**Pull request by leopoldjuergen**
Details:
- Add PortManager class.
- Add Port class.
- Add PortManager to Adapter class.
- Add Adapter to documentation reference.
- Add Port to example (example8.py).
- Add 'class' if available to __str__ of BaseResource.
- Add Unit Tests for PortManager and Port.

Signed-off-by: Juergen Leopold <leopoldj@de.ibm.com>

### CODE REVIEW

File: zhmcclient/_adapter.py [hunk 1]
```diff
@@ -194,6 +196,19 @@ def __init__(self, manager, uri, properties):
         """
         assert isinstance(manager, AdapterManager)
         super(Adapter, self).__init__(manager, uri, properties)
+        self._ports = None
+
+    @property
+    @_log_call
```
Comment (andy-maier) [inline zhmcclient/_adapter.py]: While technically, this is part of the external API, we did not put the `@log_call` decorator on any of the other properties that are managers. Suggest to remove `@log_call`.

**Comment by andy-maier** [comment]
+1, except for my comment on the `@log_call` decorator on `ports`.

**Comment by leopoldjuergen** [comment]
+1

**Comment by leopoldjuergen** [comment]
Ready for review and merge.

**Comment by andy-maier** [comment]
+1

**Pull request merged**